src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -627,7 +646,10 @@ class module_elf : public module_impl
           throw std::runtime_error("Invalid offset " + std::to_string(offset));
 
         std::string argnm{ symname, symname + std::min(strlen(symname), dynstr->get_size()) };
-        std::string key_string = generate_key_string(argnm, buf_type);
+
+        std::string key_string = (is_pure_digit(argnm)) ? // If true then argnm is argument index
+                                 std::to_string(generate_unique_number(static_cast<uint64_t>(std::stoi(argnm)), static_cast<uint64_t>(buf_type))) :
+                                 generate_key_string(argnm, buf_type);
```

Comment:
Can you please add some comments to explain why are we appending buftype (digital version) to argname? Seems a bit more complicated to me. So want to understand what benefit do we get from here.

Reply:
What benefit do we get by generating a Cantor Pairing key instead of using pure argidx?

Reply:
Use case 1:
    argname is "control-packet", without appending buftype, control-packet bo address will be used to patch instruction-buffer and patch control-packet itself. This is the root cause of VITIS-11623

Use case 2:
argname is "1", without appending buftype, ifm bo address will be used to patch instruction-buffer and control-packet regardless. But in a test case we may not need patch both